### PR TITLE
Update DDL script for role

### PR DESCRIPTION
**Pull request by kcherukupalli**
## What’s changing?

When the Role exists then alter the Role password to make sure it sync up with new client-id/app-id.   Once the password is updated it will be coupled with Azure function.   This will avoid deleting and recreating the database after recreating Azure functions.

## Why?

nac-1270

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [ ] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [x] Have IaC scripts run successfully against an existing deployed environment?
    - [x] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?
Presently the issue is after recreating the Az function it will be decoupled with the database.
Fix:
It needs to re-run the IAC script to reprovision the Az function.  Updated Role creation logic with Alter role password with app-id, if already exists.  This will maintain the coupling between the Az function and database.

## Code Review Checklist:

- [ ] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [ ] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [ ] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

**Review by bgfoshay — COMMENTED** [review]
Can you outline the anticipated workflow in the "Anything else" section of the PR? Also, do we need these steps or documentation added somewhere in the repo?

**Comment by kcherukupalli** [comment]
Updated the PR with details.

**Comment by ryangeorgeharvey** [comment]
> Updated the PR with details.

I agree with Bill that this should be documented somewhere. My recommendation is to add a note to https://github.com/18F/piipan/blob/dev/docs/iac.md#notes indicating that the ddl must be rerun if a function is deleted.

File: iac/create-databases.bash [hunk 1]
```diff
@@ -83,20 +83,31 @@ create_managed_role () {
   db=$1
   role=$2
   client_id=$3
-  # XXX If the role already exists, we assume it is already associated with the
-  # client id specified.
+  # If the role already exists, we assume it is already associated with the
+  # client id specified. 
   # This isn't quite consistent with the Azure Resource Manager incremental mode
```
Comment (ryangeorgeharvey) [inline iac/create-databases.bash:88]: minor (suggestion): Is this comment block still needed?

**Comment by kcherukupalli** [comment]
> > Updated the PR with details.
> 
> I agree with Bill that this should be documented somewhere. My recommendation is to add a note to https://github.com/18F/piipan/blob/dev/docs/iac.md#notes indicating that the ddl must be rerun if a function is deleted.

Updated as per comments.

File: docs/iac.md [hunk 2]
```diff
@@ -187,6 +187,7 @@ az resource list  --tag SysType=PerStateMatchApi --query "[? resourceGroup == 'r
     1. `az rest --method DELETE --uri https://management.azure.com/subscriptions/<subscription-id>/providers/Microsoft.ApiManagement/locations/<region>/deletedservices/<apim-service-id>?api-version=2020-06-01-preview` for each relevant APIM instance.
     1. Now when you re-run `create-resources`, you should not get an `DeploymentFailed` error in `create-apim`.
 - Some Azure CLI provisioning commands will return before all of their behind-the-scenes operations complete in the Azure environment. Very occasionally, subsequent provisioning commands in `create-resources` will fail as it won't be able to locate services it expects to be present; e.g., `Can't find app with name` when publishing a Function to a Function App. As a workaround, re-run the script.
+- If you have deleted any Function App and is re-creating, then re-run the script or run the scripts related to Alter Role which binds the Function App to database.
```
Comment (ryangeorgeharvey) [inline docs/iac.md:190]: Nitpick (minor): Change text to "If you have deleted any Function App and need to re-create

**Review by ryangeorgeharvey — APPROVED** [review]
LGTM
